### ISSUE TITLE
Unable to disable ownership check when using New-PSSession

### ISSUE THREAD

**Issue by eocron** [PowerShell/PowerShell#16136]
### Summary of the new feature / enhancement

I have some username and ssh key pair to authenticate to CentOS machine and currently unable to properly establish session through New-PSSession.

Currently I take ssh key from some other SECURE place, save it, pass path to it into New-PSSession and will dispose it after usage, but New-PSSession complains about ownership of key file. Which will be deleted anyway. For non domain user.

Let the user decide what secure and what is not. 

```
The background process reported an error with the following message: The SSH
     | client session has ended with error message:
     | @@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@ @         WARNING: UNPROTECTED PRIVATE KEY
     | FILE!          @ @@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@ Permissions for
     | 'F:\\Development\\tmp\\private.key' are too open. It is required that your private key files are NOT
     | accessible by others. This private key will be ignored. Load key "F:\\Development\\tmp\\private.key":
```

And now Im going through world of pain of granting Ownership to non-domain user name, like "foobar".
So basically New-PSSession not working for me.

### Proposed technical implementation details (optional)

Add -Ignore parameter or explicit -KeyBody or something.

**Comment by jhoneill** [comment]
I'm not an SSH expert but I've seen that message before and it comes from SSH, and is passed back to `new-pssession` , it's not generated by `new-PsSession` itself.  
My guess is that "F:\Development\tmp\": has read-write access for everyone.   

As a troubleshooting tip I've found it is worth trying to SSH to the box I want to use first and only when I know that is working, do I try to make psSession to it.

**Comment by eocron** [comment]
Im hoping that New-PSSession at least try to do this ACL nightmare, because currently I see everywhere big chunks of code which dance around ACL in hope it will grant ownership to file. And all this nightmare is just to bypass some business logic inside SSH.

**Comment by jhoneill** [comment]
Yeah, the Unix permissions are very simple.  But Changing file permissions as a side effect of running another command is not the way to go. Is there a reason you can't put the file in the users home directory where ssh should be happy with the permissions ?

**Comment by jborean93** [comment]
I would be -1 on such a thing. If ssh is complaining that the key is not protected enough you should fix that problem. Granted it is more complex on Windows to do such a thing but it shouldn't mean you leave a private key left open on the file system.

It's the same with `TrustedHosts` on WSMan, PowerShell just uses the provider and you still need to trust the connection outside of PowerShell for it to start working.

**Comment by eocron** [comment]
Not checkin CA or other things also a security risk. We just accept it.

**Comment by jhoneill** [comment]
@eocron  you are probably not the first unix person who struggles with the idea that there are more than 3 possible permissions on a file, and more than one user or one group can have access etc.  However ACLs where there in OS/2's HPFS file system, and 30 years on you're unlikely to get anyone from the Microsoft side of the fence to accept groups for Role-Based Access Control, or a common method for setting permissions on files, directory objects, or configuration settings or anything else is actually a bad thing.  So less of "ACL for windows is garbage" might help you. 

If you could make an ssh connection to your desired target OUTSIDE PowerShell remoting but it failed in PowerShell it's fair enough to ask PowerShell to fix it.  But you've put the keyfile in a windows folder which won't be acceptable to ssh - so it's not the PowerShell function which is unsuitable - you've ensured you cannot use ssh for anything. 

`icacls "F:\Development\tmp\private.key":  /grant 'yourusername:(F)' /inheritance:r `     
should keep SSH happy whether it is run from powershell or anywhere else

Before icacls there was cacls which I think goes back to early Windows NT and possibly OS/2 and there are PowerShell get- and set- -acl commands. So people have been setting acls from the command line and scripts for decades. Is it so hard for your script to set an acl  when it copies a file or put the file somewhere that inherited permissions are acceptable to ssh ?

**Comment by eocron** [comment]
Your solution is not working, nothing happens with ownership, nor it deletes all others. Nor it accepts non-domain "foobar" - which windows try to find in active directory, but we have no possibility to tell it stop this and just grant this "string" a permission, like chown does.

**Comment by jhoneill** [comment]
1. The owner will be the account which created the file. 
2. Write the account name as domain\user  where domain is "machine name" for local accounts.  

icacls on its own gives enough help that it shouldn't take you more than a few minutes
![image](https://user-images.githubusercontent.com/16209593/134433805-8102e3e0-d66c-4c1f-96dc-76557b385ac8.png)

But I think if you use $keyfile =  "$env:temp\keyfile"  you won't need to set the acl at all.

**Comment by jborean93** [comment]
Remember that it's more than just the owner, `ssh` also takes into consideration the DACL and who has read access. IIRC you also need to make sure that no user apart from yourself, `Administrators`, and `SYSTEM` have any access to the file. This applies to both permissions inherited and explicitly set on the file.
